### PR TITLE
Add E2E tests for the Active Filters Block

### LINKED ISSUE #6377: Active Filters Block: Implement E2E tests
The Active Filters block is now compatible with PHP templates and block templates using the PHP rendered `Classic Template` block. Instead of fetching and rerendering the filtered products like the All Products block does, when PHP rendered products are being filtered they are done so via URL params. We need to implement E2E tests for this behaviour:

* When the filter is applied this is identified by the Active Filters block
* When clicking the `X` on a filter it removes a filter and on PHP templates only, this triggers a page refresh.
* After the page refreshes, the URL is what we expect.
* When clicking the "Clear all" this will remove all active filters and on PHP templates only, this triggers a page refresh.

### PR DESCRIPTION
**Pull request by gigitux**
This PR adds E2E tests for the Filter Products By Attribute Block. These tests check that:
- When the filter is applied this is identified by the Active Filters block
- When clicking the `X` on a filter it removes a filter and on PHP templates only, this triggers a page refresh.
- After the page refreshes, the URL is what we expect.
- When clicking the "Clear all" this will remove all active filters and on PHP templates only, this triggers a page refresh.

Fixes #6377

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [x] E2E tests

#### User Facing Testing

Check out this branch

1. Run `npm run test:e2e tests/e2e/specs/shopper/active-filters.test.ts`.
2. Be sure that tests don't fail.

* [x] Do not include in the Testing Notes

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### CODE REVIEW

File: tests/e2e/fixtures/fixture-loaders.js [hunk 1]
```diff
@@ -455,9 +455,13 @@ const createProductAttributes = ( fixture = fixtures.Attributes() ) => {
...
-					return WooCommerce.get( 'products/attributes' ).then(
-						( response ) => response.data
-					);
+					return WooCommerce.get( 'products/attributes' )
+						.then( ( response ) => response.data )
+						.then( ( data ) => {
+							return data.find(
+								( item ) => item.name === attribute.name
+							);
+						} );
```
Comment (tomasztunik) [inline tests/e2e/fixtures/fixture-loaders.js:464]: what's the reason behind this change? doesn't seem to be used by the new tests

Comment (tomasztunik) [reply tests/e2e/fixtures/fixture-loaders.js:464]: Ok, it's the fixtures loading. If we want to return all attributes wouldn't it be better to just do one call at the end?

File: tests/e2e/specs/shopper/active-filters.test.ts [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+ * Click a link and wait for the page to load.
+ *
+ * @param {string} selector The CSS selector of the link to click.
+ */
+const clickLink = async ( selector: string ) => {
+	await Promise.all( [
+		page.click( selector ),
+		page.waitForNavigation( { waitUntil: 'networkidle0' } ),
+	] );
+};
```
Comment (tomasztunik) [inline tests/e2e/specs/shopper/active-filters.test.ts]: should we move it to utils higher up? Was looking at those as you adopted the pattern and I was like aaaa - we need an util for that so kudos for bringing it in :)

File: tests/e2e/specs/shopper/filter-products-by-price.test.ts [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+			expect( isRefreshed ).not.toBeCalled();
+
+			await setMaxPrice();
+
+			await Promise.all( [
+				page.click( selectors.frontend.submitButton ),
+				page.waitForNavigation( {
+					waitUntil: 'networkidle0',
+				} ),
+			] );
```
Comment (tomasztunik) [inline tests/e2e/specs/shopper/filter-products-by-price.test.ts]: use helper here

File: tests/e2e/fixtures/fixture-loaders.js (hunk 1 above)
Comment (gigitux) [reply tests/e2e/fixtures/fixture-loaders.js:464]: I didn't write these lines, but this code is executed only when there is an error during the POST request.

**Review by tomasztunik — APPROVED** [review]
